spec/std/isa/csr/scountinhibit.yaml

```
@@ -0,0 +1,1181 @@
+      - id: unimplemented-hpmcounter30
+        normative: false
+        when:
+          param:
+            name: COUNTINHIBIT_EN
+            index: 30
+            value: false
+        text: |
+          Since hpmcounter30 is not implemented, this field is read-only zero.
+    type(): |
+      return COUNTINHIBIT_EN[30] ? CsrFieldType::RW : CsrFieldType::RO;
+    reset_value(): |
+      return COUNTINHIBIT_EN[30] ? UNDEFINED_LEGAL : 0;
+    sw_write(csr_value): |
+      if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b0) {
+        if (mode() == PrivilegeMode::M) {
+          raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+        }
+      } else if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b1) {
+        if ((mode() == PrivilegeMode::VS) || (mode() == PrivilegeMode::VU)) {
+          raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+        }
+      }
+      CSR[mcountinhibit].HPM30 = csr_value.HPM30;
+      return csr_value.HPM30;
+  HPM31:
+    location: 31
+    alias: mcountinhibit.HPM31
+    description:
+      - id: csr-field-hpmcounter31.COUNT-effect
+        normative: true
+        when:
+          param:
+            name: COUNTINHIBIT_EN
+            index: 31
+            value: true
+        text: |
+          When set, `hpmcounter31.COUNT` does not increment; when 0, `hpmcounter31.COUNT` increments normally.
+      - id: unimplemented-hpmcounter31
+        normative: false
+        when:
+          param:
+            name: COUNTINHIBIT_EN
+            index: 31
+            value: false
+        text: |
+          Since hpmcounter31 is not implemented, this field is read-only zero.
+    type(): |
+      return COUNTINHIBIT_EN[31] ? CsrFieldType::RW : CsrFieldType::RO;
+    reset_value(): |
+      return COUNTINHIBIT_EN[31] ? UNDEFINED_LEGAL : 0;
+    sw_write(csr_value): |
+      if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b0) {
+        if (mode() == PrivilegeMode::M) {
+          raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+        }
+      } else if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b1) {
+        if ((mode() == PrivilegeMode::VS) || (mode() == PrivilegeMode::VU)) {
+          raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+        }
+      }
+      CSR[mcountinhibit].HPM31 = csr_value.HPM31;
+      return csr_value.HPM31;
+
+sw_read(): |
+  if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b0) {
+    if (mode() == PrivilegeMode::M) {
+      raise(ExceptionCode::IllegalInstruction, mode(), $encoding);
+    }
+  } else if (implemented_csr?(CsrName::menvcfg) && CSR[menvcfg].CDE == 1'b1) {
+    if ((mode() == PrivilegeMode::VS) || (mode() == PrivilegeMode::VU)) {
+      raise(ExceptionCode::VirtualInstruction, mode(), $encoding);
+    }
+  }
+  return ($bits(CSR[mcountinhibit]) & ($bits(CSR[scounteren]) & ~32'd2)) ;
```

I think this should be:

```suggestion
  return $bits(CSR[mcountinhibit]) & $bits(CSR[mcounteren]);
```

Which will return everything in mcountinihibit that has been delegated to S.